pkg/controller/volume/pvcprotection/pvc_protection_controller.go

```diff
@@ -247,7 +248,7 @@ func (c *Controller) pvcAddedUpdated(obj interface{}) {
 	}
 	glog.V(4).Infof("Got event on PVC %s", key)
 
-	if (!volumeutil.IsPVCBeingDeleted(pvc) && !volumeutil.IsProtectionFinalizerPresent(pvc)) || (volumeutil.IsPVCBeingDeleted(pvc) && volumeutil.IsProtectionFinalizerPresent(pvc)) {
+	if (pvc.ObjectMeta.DeletionTimestamp == nil && !slice.ContainsString(pvc.ObjectMeta.Finalizers, volumeutil.PVCProtectionFinalizer, nil)) || (pvc.ObjectMeta.DeletionTimestamp != nil && slice.ContainsString(pvc.ObjectMeta.Finalizers, volumeutil.PVCProtectionFinalizer, nil)) {
```

Comment:
Would it be worth it to have functions for the conditions, since the conditions are also checked in `processPVC`?  That way, if we have to change these conditions in the future, we won't forget about changing them above either.

Reply:
Yeah, it's a good point. I moved the conditions into separate functions.